#457 ffprobe stdout parsing breaks for displaymatrix

makepanic opened the pull request:
This change introduces an additional `ffprobe()` argument that is currently used to enable usage of the ffmpeg `-print_format json` option.

A couple open questions:
- i added a new test fixture from http://dev.exiv2.org/boards/3/topics/1189?r=1207#message-1207 which contains the displaymatrix bug from https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/issues/457 . On the exiv2 sample page is no license mentioned. Should i remove the `IMG_0034.MOV` fixture?
- are there better names for the `ffprobeJSONFormat` property or is it ok?
- should it be possible for the 3 args `ffprobe()` call to include the options array and the `ffprobeJSONFormat` object? The code currently only supports index + ( options array or `ffprobeJSONFormat` object)

njoyard commented:
Thanks for this, sorry for the lack of news...

A few remarks/answers:
* The licensing issue is a non-issue IMO. Those are purposefully recorded technical samples. If we ever get a removal request, I'll remove it.
* I would prefer `useJSON` or `json` for the name, no need to be *that* verbose.
* I'm not fond of so many argument syntaxes. Maybe we could get rid of the options array and have the following calling conventions:
  * ffprobe(callback)
  * ffprobe(index, callback)
  * ffprobe({...}, callback) with possible keys in the object:
    * 'json' to use JSON output
    * 'options' to pass additional command line options
    * 'index' to specify the input index

(would that be okay for you @oohnoitz ?)

oohnoitz commented:
@njoyard sounds good for me.

makepanic commented:
I updated the PR based on your feedback.
I added a logger warning for ffprobe calls with the signatures [`ffprobe(index, [...options], cb)`](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/pull/476/files#diff-7e540096be92d6f58625f3ef059640edR115) and [`ffprobe([...options], cb)`](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/pull/476/files#diff-7e540096be92d6f58625f3ef059640edR123), because removing them would result in a breaking change.
The flag to enable ffprobe json output is now called `json` and passed inside the ffprobe config object `ffprobe({json: true}, cb)`.

njoyard commented:
Nice, thanks !
Will review in a few hours :)

njoyard commented:
Looks good to me, will add a few details to documentation before merging, though.
